kong/plugins/apifortress/fortress_serializer.lua

```diff
@@ -0,0 +1,117 @@
+-- The serializer. Builds a data structure that is compatible with
+-- the API Fortress API expectations
+
+local stringy = require "stringy"
+
+local _M = {}
+
+-- pre-indexing global functions
+local str_startswith = stringy.startswith
+local str_endswith = stringy.endswith
+local str_sub = string.sub
+local str_len = string.len
+local table_insert = table.insert
+local encode_base64 = ngx.encode_base64
+
+function _M.serialize(ngx)
+
+  -- pre-indexing to reduce the number of lookups later on
+  local ngx_req = ngx.req
+  local ngx_resp = ngx.resp
+  local ngx_var = ngx.var
+  local ngx_ctx = ngx.ctx
+
+  local contentType =  nil
+  local request_headers = {}
+
+  local status = ngx.status
+  local method = ngx_req.get_method()
+  local reqContentType = nil
+
+  local reqBody = ngx_req.get_body_data()
+  local requestUri = ngx_var.request_uri
+  local uri = ngx_ctx.api.upstream_url
+
+  -- Adjusting the URI ending character to be chained with requestUri
+  if str_endswith(uri,"/") and str_startswith(requestUri, "/") then
+    uri = str_sub(uri,0,str_len(uri)-1)
+  end
+
+  -- This should represent the whole URL
+  uri = uri .. requestUri
+
+  local postBody = nil
+  local putBody = nil
```

Comment:
tip: you can be more concise here since Lua supports multiple declaration/assignment (and the assignment defaults to `nil`)
style: also our codebase favors snake case instead of camel case (in particular for variables, sometimes for methods too but that would be because of legacy code)
```lua
local body_post, body_put
```